### PR TITLE
Step 1841 errors

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Adding more context to error messages printed to the build log.

### Changes

- Top-level input parsing errors got a new format: `Process config: %s`
- Top-level step run errors got a new format: `Run: %s`
- Improved error messages
  - use lower case words in error messages
  - use `%w` for wrapping errors

Non-functional changes:
- Partial Ideal Step refactor:
  - main function is split into: `SchemeGenerator.ProcessConfig` and `SchemeGenerator.Run`

### CODE REVIEW

**Review by tothszabi — DISMISSED** [review]
🌟 for restructuring the `main.go` file

File: project.go [hunk 1]
```diff
@@ -22,12 +22,12 @@ type projectContainer struct {
 
 func newProject(path string) (projectContainer, error) {
 	if !xcodeproject.IsXcodeProj(path) {
-		return projectContainer{}, fmt.Errorf("path (%s) is not a Project", path)
+		return projectContainer{}, fmt.Errorf("%s is not an xcode project", path)
```
Comment (lpusok) [inline project.go]: xcode -> Xcode officially I beleive

File: step.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	fmt.Println()
+	log.Infof("Generating Schemes...")
+
+	projects, missingProjects, err := container.projects()
+	if err != nil {
+		return fmt.Errorf("getting projects failed: %w", err)
+	}
+
+	for _, missingProject := range missingProjects {
+		log.Warnf("Skipping Project (%s), as it is not present", pathRelativeToWorkspace(missingProject, cfg.ContainerPath))
```
Comment (lpusok) [inline step.go]:
```suggestion
		log.Warnf("Skipping project (%s), as it is not present", pathRelativeToWorkspace(missingProject, cfg.ContainerPath))
```

File: project.go (hunk 1 above)
Comment (godrei) [reply project.go]: intentionally made everything lower case. Do you think we should use `Xcode` in any case?

Comment (lpusok) [reply project.go]: I prefer the 'correct' uppercase capitalization.

**Review by lpusok — DISMISSED** [review]

File: project.go (hunk 1 above)
Comment (godrei) [reply project.go]: Updated

**Review by lpusok — APPROVED** [review]

**Pull request merged**